src/Umbraco.Infrastructure/Composing/HostBuilderExtensions.cs

```diff
@@ -1,19 +1,24 @@
-﻿using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
 
 namespace Umbraco.Core.Composing
 {
     /// <summary>
-    /// Extends the <see cref="IHostBuilder"/> to enable Umbraco to be used as the service container.
+    /// Extends the <see cref="IHostBuilder"/> to add CoreRuntime as a HostedService
     /// </summary>
     public static class HostBuilderExtensions
     {
         /// <summary>
-        /// Assigns a custom service provider factory to use Umbraco's container
+        /// Adds CoreRuntime as HostedService
         /// </summary>
-        /// <param name="builder"></param>
-        /// <returns></returns>
+        /// <remarks>
+        /// Should be called before ConfigureWebDefaults.
```

Comment:
Ah, it doesn't. I should have cleaned up better, new commit incoming.

This branch started with CoreRuntime added to the service collection just like everything else in Startup.ConfigureServices. Everything was fine for the site, CoreRuntime would start before the web host.

This was a problem for the test setup however because Startup.Configure is where we configured UmbracoDatabaseFactory to use one of the test databases, this would happen after CoreRuntime started and determined run level. RunLevel would always be install during tests when components were initialized.

So i moved the hosted service registration into the host builder extension and called it after UseWebHost, that way Startup.Configure comes before CoreRuntime start, good for tests bad for website e.g. install routes which are conditionally added based on run level.

I had a while where I would call HostBuilder.UseUmbraco first in website and last in tests, this worked but is obviously weird as fuck so I knew I had to come back and fix it.

The fix is simple, just call UseTestDatabase before IHost start, that way we can use the same order in tests and when running the site.

I cleaned up to remove odd instructions around tests but really I should have nuked the whole thing.